### ISSUE TITLE
Snowflake dialect: AM02 `ambiguous.union` should not be enabled

### ISSUE THREAD

**Issue by hchasestevens** [sqlfluff/sqlfluff#6469]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running `sqlfluff --dialect snowflake`, bare `UNION`s are flagged by the `ambiguous.union` rule:

```bash
$ cat test.sql
SELECT a.x FROM foo AS a
UNION
SELECT b.x FROM bar AS b

$ sqlfluff lint --dialect snowflake test.sql
== [test.sql] FAIL
L:   2 | P:   1 | AM02 | 'UNION [DISTINCT|ALL]' is preferred over just 'UNION'.
                       | [ambiguous.union]
All Finished 📜 🎉!
```

### Expected Behaviour

`ambiguous.union` should not be enforced for the Snowflake dialect; `UNION` in Snowflake should be permitted.

### Observed Behaviour

The rule requires that bare `UNION` be replaced by `UNION ALL` or `UNION DISTINCT`. Per https://docs.snowflake.com/en/sql-reference/operators-query#union-all , Snowflake does not support `UNION DISTINCT`, only `UNION` and `UNION ALL` - with these two having different behavior.

### How to reproduce

SQL:
```sql
SELECT a.x FROM foo AS a
UNION
SELECT b.x FROM bar AS b
```

Command:
`sqlfluff lint --dialect snowflake test.sql`

### Dialect

Snowflake

### Version

3.2.5

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by hchasestevens** [comment]
Per https://github.com/sqlfluff/sqlfluff/issues/5964#issuecomment-2185093577 , apparently, Snowflake does secretly support `UNION DISTINCT`. On that basis, feel free to close this ticket, although I don't think it's great that we're assuming users will infer that they're meant to adopt undocumented behavior.

**Comment by alanmcruickshank** [comment]
Thanks for the check @hchasestevens 👍 . You're correct that this rule is enabled for snowflake because the functionality is there (and widely used), but you're also correct that the snowflake docs don't reflect that.
